File: commands/matches/stats.js
const { SlashCommandBuilder } = require('discord.js');
const { playerService } = require('../../services/playerService');

module.exports = {
  data: new SlashCommandBuilder()
    .setName('stats')
    .setDescription('View your match statistics'),

  async execute(interaction) {
    try {
      const player = await playerService.getPlayerByDiscord(interaction.user.id);

      if (!player) {
        return interaction.reply({ content: 'Player not found.', ephemeral: true });
      }

      const games = (player.wins || 0) + (player.losses || 0);
      const winRate = games > 0 ? ((player.wins / games) * 100).toFixed(1) : 0;

      return interaction.reply({
        content:
          `📊 **Your Stats**\n` +
          `Games: ${games}\n` +
          `Wins: ${player.wins || 0}\n` +
          `Losses: ${player.losses || 0}\n` +
          `Win Rate: ${winRate}%`,
        ephemeral: true
      });

    } catch (err) {
      console.error('❌ Stats command error:', err);
      return interaction.reply({ content: 'Error fetching stats.', ephemeral: true });
    }
  }
};

File: commands/matches/stats.py
from discord import app_commands
from discord.ext import commands
from services.player_service import player_service

GUILD_ID = 1335339358932304055

class Stats(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @app_commands.command(name="stats", description="View your match statistics")
    async def stats(self, interaction):
        try:
            player = player_service.get_player_by_discord(interaction.user.id)

            if not player:
                return await interaction.response.send_message(
                    "Player not found.",
                    ephemeral=True
                )

            wins = player.get("wins", 0)
            losses = player.get("losses", 0)
            games = wins + losses
            win_rate = round((wins / games) * 100, 1) if games > 0 else 0

            msg = (
                f"📊 **Your Stats**\n"
                f"Games: {games}\n"
                f"Wins: {wins}\n"
                f"Losses: {losses}\n"
                f"Win Rate: {win_rate}%"
            )

            return await interaction.response.send_message(msg, ephemeral=True)

        except Exception as e:
            print(f"❌ Stats command error: {e}")
            return await interaction.response.send_message(
                "Error fetching stats.",
                ephemeral=True
            )

async def setup(bot):
    await bot.add_cog(Stats(bot))


